code/modules/fishing/sources/source_types.dm

```diff
@@ -773,3 +773,77 @@
 	)
 	fishing_difficulty = FISHING_DEFAULT_DIFFICULTY + 20
 	fish_source_flags = FISH_SOURCE_FLAG_EXPLOSIVE_MALUS
+
+/datum/fish_source/dimensional_rift
+	catalog_description = null //it's a secret (sorta, I know you're reading this)
+	radial_state = "cursed" // placeholder
+	overlay_state = "portal_rift_2" // yeah good luck adaptin the rift sprite to this template. recolored randomizer's the best you're getting
+	fish_table = list(
+		FISHING_INFLUENCE = 6,
+		/obj/item/fish/starfish/chrystarfish = 7,
+		/obj/item/fish/dolphish = 7,
+		/obj/item/fish/flumpulus = 7,
+		/obj/item/fish/gullion = 7,
+		/mob/living/basic/heretic_summon/fire_shark/wild = 3,
+		/obj/item/eldritch_potion/crucible_soul = 1,
+		/obj/item/eldritch_potion/duskndawn = 1,
+		/obj/item/eldritch_potion/wounded = 1,
+		/obj/item/reagent_containers/cup/beaker/eldritch = 2,
+	)
+	fish_counts = list(
+		/mob/living/basic/heretic_summon/fire_shark/wild = 3,
+		/obj/item/eldritch_potion/crucible_soul = 1,
+		/obj/item/eldritch_potion/duskndawn = 1,
+		/obj/item/eldritch_potion/wounded = 1,
+		/obj/item/reagent_containers/cup/beaker/eldritch = 2,
+	)
+	fish_count_regen = list(
+		/mob/living/basic/heretic_summon/fire_shark/wild = 3 MINUTES,
+		/obj/item/eldritch_potion/crucible_soul = 5 MINUTES,
+		/obj/item/eldritch_potion/duskndawn = 5 MINUTES,
+		/obj/item/eldritch_potion/wounded = 5 MINUTES,
+		/obj/item/reagent_containers/cup/beaker/eldritch = 2.5 MINUTES,
+	)
+	fishing_difficulty = FISHING_DEFAULT_DIFFICULTY + 35
+	fish_source_flags = FISH_SOURCE_FLAG_EXPLOSIVE_NONE
```

Comment:
There are a couple other fish sources that could use this flag, chasms and aquariums iirc.